src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +30,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = address;
```

Comment:
You are still using the value String. What was the point of creating all the classes if you keep using this variable. value should be maintained only to keep track of the user input. In any other place you must now start using the classes you created.